### ISSUE TITLE
Add Google Authentication

### ISSUE THREAD

**Issue by gannonprudhomme** [aggie-coding-club/Rev-Registration#156]
This will most likely involve adding some form of Student/User model, as we'll most likely need to save their access & refresh token that is returned from OAuth

This is both a frontend and backend task since this might as well including adding the login button on the frontend (since it's not that involved)

**Comment by gannonprudhomme** [comment]
Will probably want to use [social-auth-app-django](https://github.com/python-social-auth/social-app-django). Tutorial for it [here](https://medium.com/trabe/oauth-authentication-in-django-with-social-auth-c67a002479c1).

**Comment by gannonprudhomme** [comment]
Since you'll need to store the google client ID & secret, we should put these in a `.env` file in `autoscheduler/autoscheduler/settings/.env`. We can use the [`python-dotenv`](https://pypi.org/project/python-dotenv/) pip package to do this easily. In order to integrate this, add the necessary import (`from dotenv import load_dotenv`) and `load_dotenv()` to the top of both `base.py` and `wsgi.py`.

**Comment by gannonprudhomme** [comment]
This should also connect "Google Login" button on the frontend to the login page. To do this, go in  `autoscheduler/frontend/src/components/NavBar.tsx`, then for the `Google Login` button, add an `onClick` that runs `window.location.href = 'login/google-oauth2/'`. (Emphasis on that last slash, without it it won't redirect correctly). We might want to make this pop up into its own window (like what aggiescheduler does) rather than redirect to a new page, but I think this is fine for now. 

I also already set up everything for GCP in regards to OAuth, so you'll just need to go to APIs & Services of GCP console (http://console.cloud.google.com/apis/credentials), then get the client ID & secret from there.

I've implemented almost all of this functionality locally, so let me know if you have questions with any of it.
